### PR TITLE
Add script to help with troubleshooting some dev environment issues

### PR DESCRIPTION
**Pull request by donokuda**
A common thing that I've been running into often is my development environment not working if I haven't been keeping up to date with changes regularly. When I would pull a branch and try building, I'd often run into errors similar to this:

screen shot 2018-12-13 at 10 06 56 am

Some of the engineers suggested closing Visual Studio and running the following commands:

```bash
git clean -xdff
git submodule foreach git clean -xdff
```

Which ends up solving most of the issues I've been running into! 🎉 

I've added a script (`scripts/clean.ps1`) and took a crack at [documenting when to use this command](https://github.com/github/VisualStudio/tree/clean-me#troubleshooting) in the README, however there's a chance it could be improved a bit.

I've also added `git submodule update` to the top of the ~`clean.ps1`~ `clean.cmd` script so that submodules such as Octokit are sync'd before running the command.

/cc @grokys @StanleyGoldman since y'all have helped me this week regarding my dev issues.

### CODE REVIEW

File: clean.cmd [hunk 1]
```diff
@@ -0,0 +1 @@
+powershell -ExecutionPolicy Unrestricted scripts\clean.ps1
```
Comment (grokys) [inline clean.cmd]: Is there any reason why this script needs to call out to `powershell` to run the git commands? Couldn't it just run them here?

Comment (donokuda) [reply clean.cmd]: You're right! I was following existing examples involving `.cmd` and `.ps1` files. I've moved things over to the `clean.cmd` flle and 🔥 the `clean.ps1`

**Comment by donokuda** [comment]
🛀@grokys Ready for another review. Thanks for taking a look at this 😄

**Comment by grokys** [comment]
Thanks @donokuda !

**Pull request merged**